azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -90,17 +86,16 @@
    @Resource
    @Named(ComputeServiceConstants.COMPUTE_LOGGER)
    private Logger logger = Logger.NULL;
-
+   private final Json json;
    private final AzureComputeApi api;
-
    private final AzureComputeConstants azureComputeConstants;
-
    private final ProviderMetadata providerMetadata;
+   private final DeploymentToVMDeployment deploymentToVMDeployment;
```

Comment:
Better declare and inject it as `Function<Deployment, VMDeployment>` to make it easier to re-bind.